src/core/include/openvino/op/roi_align.hpp

```diff
@@ -81,6 +81,88 @@ class OPENVINO_API ROIAlign : public Op {
     PoolingMode m_mode;
 };
 }  // namespace v3
+
+namespace v9 {
+class OPENVINO_API ROIAlign : public Op {
+public:
+    OPENVINO_OP("ROIAlign", "opset9", op::Op, 9);
+    BWDCMP_RTTI_DECLARATION;
+    enum class PoolingMode { AVG, MAX };
+    enum class AlignedMode { ASYMMETRIC, TF_HALF_PIXEL_FOR_NN, HALF_PIXEL };
+
+    ROIAlign() = default;
+    /// \brief Constructs a ROIAlign node matching the ONNX ROIAlign specification
+    ///
+    /// \param input           Input feature map {N, C, H, W}
+    /// \param rois            Regions of interest to pool over
+    /// \param batch_indices   Indices of images in the batch matching
+    ///                        the number or ROIs
+    /// \param pooled_h        Height of the ROI output features
+    /// \param pooled_w        Width of the ROI output features
+    /// \param sampling_ratio  Number of sampling points used to compute
+    ///                        an output element
+    /// \param spatial_scale   Spatial scale factor used to translate ROI coordinates
+    /// \param mode            Method of pooling - 'avg' or 'max'
+    /// \param aligned_mode    Method of coordinates alignment - 'asymmetric', 'tf_half_pixel_for_nn' or 'half_pixel'
+    ROIAlign(const Output<Node>& input,
+             const Output<Node>& rois,
+             const Output<Node>& batch_indices,
+             const int pooled_h,
+             const int pooled_w,
+             const int sampling_ratio,
+             const float spatial_scale,
+             const std::string& mode,
+             const std::string& aligned_mode = "asymmetric");
```

Comment:
Can we reduce the number of constructors and keep only constructor with enums?

Reply:
have reduced the number of constructors and kept only constructor with enums which is aligned with spec